openlane/scripts/openroad/common/report_cell_count.tcl

```diff
@@ -0,0 +1,75 @@
+# Copyright 2024 Efabless Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+proc search_multi_pattern {name patterns} {
+    foreach pattern $patterns {
+        if { [lsearch -glob $name $pattern] != -1 } {
+            return true
+        }
+    }
+    return false
+}
+
+proc report_cell_count {args} {
+    set diode_pattern [lindex [split "$::env(DIODE_CELL)" "/"] 0]
+    set physical_welltap 0
+    set physical_spacer 0
+    set physical_antennacell 0
+    set physical_misc 0
+    set inverter 0
+    set sequential 0
+    set buffer 0
+    set clock_gate 0
+    set logical 0
+    set macros 0
+    set insts [$::block getInsts]
+    foreach inst $insts {
+        set opensta_instance [get_cells [$inst getName]] ;# exposes different apis
+        set lef_master [$inst getMaster]
+        set types [split [$lef_master getType] _]
+        if { "$types" == "CORE" } {
+            if { [get_property $opensta_instance is_buffer] } {
```

Comment:
This logic implies mutual exclusivity between a lot of types. Is that true for cells?

Reply:
Think of it less like a strict separation of logical elements and more of a grouping by priority.